<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>685</width>
    <height>494</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chat Client</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QStackedWidget" name="pagesWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>661</width>
      <height>431</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="page_0">
     <widget class="QPlainTextEdit" name="ipText">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>130</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="plainText">
       <string>IP:</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="portText">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>130</y>
        <width>41</width>
        <height>31</height>
       </rect>
      </property>
      <property name="plainText">
       <string>Port:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="connectButton">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>180</y>
        <width>131</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Connect</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="errorMessageInput">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>30</y>
        <width>461</width>
        <height>31</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="ipInput">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>130</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="portInput">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>130</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page_1">
     <widget class="QPushButton" name="loginButton">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>140</y>
        <width>141</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Login</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="usernameLoginText">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>90</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="plainText">
       <string>Username:</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="passwordLoginText">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>90</y>
        <width>61</width>
        <height>31</height>
       </rect>
      </property>
      <property name="plainText">
       <string>Password:</string>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>210</y>
        <width>671</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="usernameRegisterText">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>300</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="plainText">
       <string>Username:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="registerButton">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>350</y>
        <width>141</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Register</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="passwordRegisterText">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>300</y>
        <width>61</width>
        <height>31</height>
       </rect>
      </property>
      <property name="plainText">
       <string>Password:</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="loginError">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>20</y>
        <width>511</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="registerError">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>240</y>
        <width>511</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="usernameLoginField">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>90</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="passwordLoginField">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>90</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="usernameRegisterField">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>300</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="passwordRegisterField">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>300</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2">
     <widget class="QPushButton" name="messageSend">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>302</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Send</string>
      </property>
     </widget>
     <widget class="QTextBrowser" name="userList">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>30</y>
        <width>131</width>
        <height>221</height>
       </rect>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="currentRoomText">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>10</y>
        <width>131</width>
        <height>21</height>
       </rect>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="plainText">
       <string>Lobby</string>
      </property>
      <property name="backgroundVisible">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="logoutButton">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>370</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Logout</string>
      </property>
     </widget>
     <widget class="QScrollArea" name="scrollArea">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>451</width>
        <height>249</height>
       </rect>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>449</width>
         <height>247</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="chatBox">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
     </widget>
     <widget class="QTextBrowser" name="emojiBox">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>370</y>
        <width>361</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="messageInput">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>290</y>
        <width>371</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="openNewRoomButton">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>260</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Join Room</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page">
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>60</y>
        <width>491</width>
        <height>261</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <kerning>true</kerning>
       </font>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="joinRoomButton">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>150</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Join Room</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="newRoomField">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>110</y>
         <width>221</width>
         <height>21</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>685</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
